### ISSUE TITLE
solve result in OverflowError: Python int too large to convert to C long

### ISSUE THREAD

**Issue by hoyeunglee** [sympy/sympy#13497]
sol1 = solve([Eq(Sol[sa] - exp(xx),0),Eq(Sol[sb] - exp(y1),0)],[xx,y1])

I should return complex number
if use Rational=False, it return empty []

Traceback (most recent call last):
  File "<stdin>", line 45, in <module>
  File "C:\Python27\lib\site-packages\sympy\solvers\solvers.py", line 1067, in solve
    solution = _solve_system(f, symbols, **flags)
  File "C:\Python27\lib\site-packages\sympy\solvers\solvers.py", line 1810, in _solve_system
    soln = _solve(eq2, s, **flags)
  File "C:\Python27\lib\site-packages\sympy\solvers\solvers.py", line 1423, in _solve
    poly = Poly(f_num)
  File "C:\Python27\lib\site-packages\sympy\polys\polytools.py", line 129, in __new__
    return cls._from_expr(rep, opt)
  File "C:\Python27\lib\site-packages\sympy\polys\polytools.py", line 240, in _from_expr
    return cls._from_dict(rep, opt)
  File "C:\Python27\lib\site-packages\sympy\polys\polytools.py", line 189, in _from_dict
    return cls.new(DMP.from_dict(rep, level, domain), *gens)
  File "C:\Python27\lib\site-packages\sympy\polys\polyclasses.py", line 260, in from_dict
    return cls(dmp_from_dict(rep, lev, dom), dom, lev)
  File "C:\Python27\lib\site-packages\sympy\polys\densebasic.py", line 1011, in dmp_from_dict
    for k in range(n, -1, -1):
OverflowError: Python int too large to convert to C long

**Comment by hoyeunglee** [comment]
Typo, it should return complex number but error when run

**Comment by oscarbenjamin** [comment]
What is `Sol` in this example?

When I run this I get:
```julia
In [1]: sol1 = solve([Eq(Sol[sa] - exp(xx),0),Eq(Sol[sb] - exp(y1),0)],[xx,y1])                                                   
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-1-96d3831950ce> in <module>
----> 1 sol1 = solve([Eq(Sol[sa] - exp(xx),0),Eq(Sol[sb] - exp(y1),0)],[xx,y1])

NameError: name 'Sol' is not defined
```

**Comment by hoyeunglee** [comment]
I lost the script due to my notebook leak electricity and long time not used, you can guess an example to have complex number as solution then try this in sympy then you will know.

**Comment by oscarbenjamin** [comment]
Hi @hoyeunglee can you be more specific? i'm sorry I don't understand your explanation.

Can you show an example of code that produces this error?

**Comment by hoyeunglee** [comment]
![image](https://user-images.githubusercontent.com/5094741/53632668-86e95680-3c50-11e9-8443-2bf4878698b5.png)

i did not remember the equation i used in the past.

i just tried 
from sympy import *
xx, y1 = symbols('xx y1')
sol1 = solve([Eq(y1 - exp(xx),0),Eq(xx - exp(y1),0)],[xx,y1])

**Comment by oscarbenjamin** [comment]
Okay well that's a very different error message.

Do you think that SymPy should be able to solve this?

[Wolfram](https://www.wolframalpha.com/input/?i=solve+y+-+exp(exp(y))+for+y) only finds numerical solutions. SymPy can also find those:
```
In [5]: nsolve([y-exp(x), x-exp(y)], [x, y], [I, I])
Out[5]: 
⎡0.318131505204764 + 1.33723570143069⋅ⅈ⎤
⎢                                      ⎥
⎣0.318131505204764 + 1.33723570143069⋅ⅈ⎦

In [6]: nsolve([y-exp(x), x-exp(y)], [x, y], [-I, -I])
Out[6]: 
⎡0.318131505204764 - 1.33723570143069⋅ⅈ⎤
⎢                                      ⎥
⎣0.318131505204764 - 1.33723570143069⋅ⅈ⎦
```

**Comment by hoyeunglee** [comment]
i can not search and find my old code in old notebook, i will use nsolve instead of solve in future when meet again, thank you.

**Comment by hoyeunglee** [comment]
please close case.

**Comment by oscarbenjamin** [comment]
Okay thanks for checking this. I'll close this now but please let us know if you have other problems in future.
